datamade/la-metro-councilmatic
# Initialize load testing script, increase number of gunicorn workers

## Overview

See title. Based on [the results of load testing](https://github.com/datamade/la-metro-councilmatic/issues/585#issuecomment-616810528), this PR also:

- Increases the number of [Gunicorn workers](https://docs.gunicorn.org/en/stable/design.html#how-many-workers) from 1 to 2 on staging, and 3 on production. 
- As the Gunicorn workers are configured to respawn after 1,000 requests, it also adds [a jitter](https://docs.gunicorn.org/en/stable/settings.html#max-requests-jitter) of 100, so the workers don't respawn all at once.
- Adds a timeout of 120 to the staging instance, to bring it inline with production.

### Checklist


## Testing Instructions

 * On your local machine, start Locust: `docker-compose -f docker-compose.yml -f docker-compose.locust.yml run --rm --service-ports locust`
 * Start a test with the following parameters:

    ```
    Number of users: 325
    Hatch rate: 3
    Host: https://lametro-upgrade.datamade.us
    ```

    Confirm the median response time remains < 2000ms.

Handles #585.
